### ISSUE TITLE
Slow phpunit execution

### ISSUE THREAD

**Issue by shyim** [DEVSENSE/phptools-docs#725]
Hey,

when I press here on the run, how does it look like the PHPUnit command you execute?

It seems to parse all tests and then execute one and this takes longer than expected

<img width="429" alt="image" src="https://github.com/user-attachments/assets/61bc5a5f-69a6-4e72-a726-bd5e2120c0b7">

Could you change it too:
```
vendor/bin/phpunit <FILE-PATH> --filter METHODNAME
```

**Comment by leoasis** [comment]
This is a problem I'm facing too, we have a huge amount of tests, so not passing the file path for the test I want to run causes PHPUnit to load the entire test suite which can take a long time.

**Comment by jakubmisek** [comment]
Please see the output in VSCode's OUTPUT window, "PHP (PHPUnit)" tab. There will be a full command line being executed. We are using `--filter` to match just the specific test, but please check the actual command  if it's what is expected.

**Comment by leoasis** [comment]
this is the command I see when it's run:

```sh
"php" -d display_errors=on "/.../vendor/bin/phpunit" -c phpunit.xml --teamcity --colors=never --filter "(^(Namespace.For.The.Test::testIWantToRun(\s.*)?)$)"
```

Note that the command is loading the _entire_ test suite, and filtering the test class by namespace. Because we have a huge test suite, loading it all only to execute a single test file adds a lot of time, even multiple minutes (we have thousands of files and test files).

Compare this to this other extension (which I'm currently using to execute the tests instead), which specifies the test file path, so that's the only test file loaded:

```sh
php vendor/bin/phpunit --filter=^.*::(testIWantToRun)( with data set .*)?$ /path/for/the/test.php --colors=never --teamcity
```

This command runs instantly.

**Comment by jakubmisek** [comment]
From the comments at https://stackoverflow.com/questions/26095051/how-to-run-single-test-method-with-phpunit
we can explicitly specify the method name, class name, and file name, so PHPUnit will load only that.

I'll try to make the changes and prepare a pre-release.

**Comment by jakubmisek** [comment]
Thank you @leoasis - the `--filter` argument has been improved so whenever possible, we pass the file path there (even for selection of tests/groups in the same file)

This feature will be available right in the next release or pre-release.

**Comment by leoasis** [comment]
nice, thank you @jakubmisek for the fast turnaround!

**Comment by jakubmisek** [comment]
@leoasis thanks for pointing this out in the first place. We've just published full release 1.54 with the feature and more.

**Comment by leoasis** [comment]
I just updated the extension and I can verify now tests run immediately, thanks again!
